File: tauri/src/models/summary.ts
export interface ConversationSummary {
  id: string;
  session_id: string;
  workspace_id: string;
  summary_type: 'info' | 'extensive';
  content: string;
  key_topics: string[];
  message_range_start: number;
  message_range_end: number;
  token_count: number;
  created_at: string;
  updated_at: string;
}

File: tauri/src-tauri/src/models/summary.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConversationSummary {
    pub id: String,
    pub session_id: String,
    pub workspace_id: String,
    pub summary_type: String,
    pub content: String,
    pub key_topics: String,
    pub message_range_start: i32,
    pub message_range_end: i32,
    pub token_count: i32,
    pub created_at: String,
    pub updated_at: String,
}
